### PR TITLE
Add episode 33

### PR DESCRIPTION
**Pull request by bitsondatadev**

### CODE REVIEW

File: _episodes/33.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+Trino 370
+
+* Add CLI support for ARM64
+* Improved performance for ORC
+* Improved performance for map and row types
+* Reduced latency for OAuth2.0 authentication
+
+Trino 371
+
+tbd
```
Comment (mosabua) [inline _episodes/33.md:58]: * Support for secrets and user group selector in resource group manager.
* Support AWS role session name in S3 security mapping configuration.
* Many bug fixes.

File: _episodes/33.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+* Improved performance for map and row types
+* Reduced latency for OAuth2.0 authentication
+
+Trino 371
+
+* Support for secrets and user group selector in resource group manager.
+* Support AWS role session name in S3 security mapping configuration.
+* Many bug fixes.
+
+Notes from Manfred
```
Comment (mosabua) [inline _episodes/33.md:62]: Add end of sentence . in all items in the following list .. I missed some

File: _episodes/33.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+* A bunch of SPI changes
+* DML with Iceberg connector with fault tolerant mode and more Tardigrade improvements
+* Drop support for Kudu 1.13.0
+  
+More detailed information is available in the [Trino
+369](https://trino.io/docs/current/release/release-369.html), [Trino
+370](https://trino.io/docs/current/release/release-370.html), and [Trino
+371](https://trino.io/docs/current/release/release-371.html) release notes.
+
+## Concept of the month: High Availability with Trino
```
Comment (mosabua) [inline _episodes/33.md:87]: availability

File: _episodes/33.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+Goldman Sachs uses Trino to reduce last-mile ETL and provide a unified way of
```
Comment (mosabua) [inline _episodes/33.md:89]: ETL, and

File: _episodes/33.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+369](https://trino.io/docs/current/release/release-369.html), [Trino
+370](https://trino.io/docs/current/release/release-370.html), and [Trino
+371](https://trino.io/docs/current/release/release-371.html) release notes.
+
+## Concept of the month: High Availability with Trino
+
+Goldman Sachs uses Trino to reduce last-mile ETL and provide a unified way of 
+accessing data through federated joins. Making a variety of data sets from 
+different sources available in one spot for our data science team was a tall 
+order. Data must be quickly accessible to data consumers and systems like Trino
```
Comment (mosabua) [inline _episodes/33.md:92]: consumers, and systems

File: _episodes/33.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+must be reliable for users to trust this singular access point for their data.
+
+In order for analysts and data scientists to use these services, they first need
+to trust in the system. It was vital to Goldman Sachs that Trino have High
```
Comment (mosabua) [inline _episodes/33.md:96]: Trino has high availability. In the event of any failure, another Trino cluster is available to process requests.

File: _episodes/33.md [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+Availability so that in the event of failure, there was always another Trino
+cluster available to take requests.
+
+### Integrating Trino into the Goldman Sachs internal ecosystem 
+
+Before high availability was a concern, the team had to first integrate Trino to
+meet their requirements. This included integrating with internal security 
+systems, observability systems, and credential stores. It also meant
+adding integration with their governance services that manage cataloguing
+services and data discovery engines. Finally, while many of the connectors that
```
Comment (mosabua) [inline _episodes/33.md:106]: Finally, while many of the Trino connectors that the team intended to use exist, there were many ...

File: _episodes/33.md [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+
+Once the team had much of Trino running for some initial use cases, the next 
+step was to improve support for more simultaneous use cases and highly 
+concurrent workloads. The team wanted trust in the system and so as they scaled
+the ability to run blue-green deployments, enable resources isolation, and
+have highly available clusters through failures became much more pertinant.
+
+### Trino ecosystem at Goldman Sachs
+
+Here is an overview of the Goldman Sachs Ecosystem. It showcases the preexisting
```
Comment (mosabua) [inline _episodes/33.md:123]: ecosystem.

File: _episodes/33.md [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+services that needed to connect to Trino, the catalogs supported, and the method
+in which Goldman Sachs acheives High Availability through supporting multiple
```
Comment (mosabua) [inline _episodes/33.md:125]: achieves 

and 

high availability

File: _episodes/33.md [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+groups.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/33/trinodynamicqueryrouting.png"/><br/>
+Source: <a href="https://developer.gs.com/blog/posts/enabling-highly-available-trino-clusters-at-goldman-sachs">Goldman Sachs Blog</a>
+</p>
+
+### Query routing components
+
+* [Envoy](https://www.envoyproxy.io/) Proxy - open source edge and service proxy
```
Comment (mosabua) [inline _episodes/33.md:146]: use

```
[Envoy Proxy](https://www.envoyproxy.io/)
```

or drop "Proxy"

File: _episodes/33.md [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+that provides features sych as routing, traffic management, load balancing,
```
Comment (mosabua) [inline _episodes/33.md:147]: such

File: _episodes/33.md (hunk 10 above)
Comment (mosabua) [reply _episodes/33.md:146]: also .. do we want to mention that this is a CNCF project?

File: _episodes/33.md [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+<img align="center" width="50%" height="100%" src="/assets/episode/33/trinocontrolplane.png"/><br/>
+Source: <a href="https://developer.gs.com/blog/posts/enabling-highly-available-trino-clusters-at-goldman-sachs">Goldman Sachs Blog</a>
+</p>
+* Cluster Groups - cluster group is a set of various Trino clusters that can
+be assigned traffic by the 
+* Cluster Metadata Service - a service that provides the Envoy routers with all
+the cluster related configurations
+* Router Service
+  * Envoy Control Plane - The Envoy Control Plane is an xDs gRPC-based service, 
+  and is responsible for providing dynamic configurations to Envoy.
```
Comment (mosabua) [inline _episodes/33.md:159]: that is responsible for

File: _episodes/33.md [hunk 13]
```diff
@@ -0,0 +1,266 @@
...
+Source: <a href="https://developer.gs.com/blog/posts/enabling-highly-available-trino-clusters-at-goldman-sachs">Goldman Sachs Blog</a>
+</p>
+* Cluster Groups - cluster group is a set of various Trino clusters that can
+be assigned traffic by the 
+* Cluster Metadata Service - a service that provides the Envoy routers with all
+the cluster related configurations
+* Router Service
+  * Envoy Control Plane - The Envoy Control Plane is an xDs gRPC-based service, 
+  and is responsible for providing dynamic configurations to Envoy. 
+  * Upstream Cluster Selection -Envoy provides HTTP filters to parse and modify
```
Comment (mosabua) [inline _episodes/33.md:160]: space before Envoy

File: _episodes/33.md [hunk 14]
```diff
@@ -0,0 +1,266 @@
...
+  both request and response headers. We use a custom Lua filter to parse the 
+  request and extract the `x-trino-user` header. Then, we call the router 
+  service, which returns the upstream cluster address.
+
+## PR of the month: PR 8956 Add support for external db for schema management in mongodb connector
```
Comment (mosabua) [inline _episodes/33.md:165]: MongoDB connector

File: _episodes/33.md [hunk 15]
```diff
@@ -0,0 +1,266 @@
...
+
+This month's [PR of the month](https://github.com/trinodb/trino/pull/8956) comes
+from today's guest Siddhant to solve [this issue related to the MongoDB connector](https://github.com/trinodb/trino/issues/8887).
+
+Siddhant created the issue in response to the common issue that MongoDB
```
Comment (mosabua) [inline _episodes/33.md:170]: to the common problem that

File: _episodes/33.md [hunk 16]
```diff
@@ -0,0 +1,266 @@
...
+  request and extract the `x-trino-user` header. Then, we call the router 
+  service, which returns the upstream cluster address.
+
+## PR of the month: PR 8956 Add support for external db for schema management in mongodb connector
+
+This month's [PR of the month](https://github.com/trinodb/trino/pull/8956) comes
+from today's guest Siddhant to solve [this issue related to the MongoDB connector](https://github.com/trinodb/trino/issues/8887).
+
+Siddhant created the issue in response to the common issue that MongoDB 
+connector users face when they don't have write capability in the Mongo system.
```
Comment (mosabua) [inline _episodes/33.md:171]: in the MongoDB system

File: _episodes/33.md [hunk 17]
```diff
@@ -0,0 +1,266 @@
...
+Since Mongo has no implicit schema, Trino uses a schema definition that is
```
Comment (mosabua) [inline _episodes/33.md:172]: MongoDB

File: _episodes/33.md [hunk 18]
```diff
@@ -0,0 +1,266 @@
...
+written to a special Mongo database. This PR enables users without write access
```
Comment (mosabua) [inline _episodes/33.md:173]: MongoDB

File: _episodes/33.md [hunk 19]
```diff
@@ -0,0 +1,266 @@
...
+to create an external location to store their schema to avoid this issue.
+
+Thanks Siddhant for raising this issue, as it's a common issue beginners using
+the MongoDB connector face commonly.
+
+## Bonus PR of the month: PR 8202 Metadata for alias in elasticsearch connector only uses the first mapping
```
Comment (mosabua) [inline _episodes/33.md:179]: Elasticsearch connector

File: _episodes/33.md [hunk 20]
```diff
@@ -0,0 +1,266 @@
...
+
+This bonus [PR of the month](https://github.com/trinodb/trino/pull/8202) comes
+from another one of today's guest Suman which ends up solving multiple issues,
```
Comment (mosabua) [inline _episodes/33.md:182]: guests, Suman. It solves multiple issues,

File: _episodes/33.md [hunk 21]
```diff
@@ -0,0 +1,266 @@
...
+indexes and documents follow this schema. This pull request addresses this issue
+by scanning a much larger sample of mappings and merging the schema to handle
+any conflicts. It then goes further to cache these merged mappings for a given
+amount of time.
+
+Thanks for all of your continued work on this Suman! It will help a lot!
+
+## Demo of the month: Trino Fiddle: A tool for easy online testing and sharing of Trino SQL problems and their solutions
+
+This months demo will showcase a tool that Brian modified from [SQL Fiddle](http://sqlfiddle.com/)
```
Comment (mosabua) [inline _episodes/33.md:204]: demo showcases a tool...

File: _episodes/33.md [hunk 22]
```diff
@@ -0,0 +1,266 @@
...
+tool called Trino Fiddle. This tool will allow Trino users to share problems
+and answer questions that other Trino users are facing.
+
+This tool is currently being tested so it's not available for public use yet,
```
Comment (mosabua) [inline _episodes/33.md:208]: I would drop this. Or reword to make sure and omit promise about future release for now...

File: _episodes/33.md [hunk 23]
```diff
@@ -0,0 +1,266 @@
...
+## Demo of the month: Trino Fiddle: A tool for easy online testing and sharing of Trino SQL problems and their solutions
+
+This months demo will showcase a tool that Brian modified from [SQL Fiddle](http://sqlfiddle.com/)
+tool called Trino Fiddle. This tool will allow Trino users to share problems
+and answer questions that other Trino users are facing.
+
+This tool is currently being tested so it's not available for public use yet, 
+but it will be released soon!
+
+## Question of the month: Does trino hive connector supports CarbonData?
```
Comment (mosabua) [inline _episodes/33.md:211]: Does Trino support CarbonData?

File: _episodes/33.md [hunk 24]
```diff
@@ -0,0 +1,266 @@
...
+
+This month's [question of the month](https://www.trinoforum.org/t/142) 
+comes from [Mahebub Sayyed](https://www.trinoforum.org/u/masayyed/summary) on 
+Trino Forum. He asks, "Does trino hive connector supports CarbonData?"
```
Comment (mosabua) [inline _episodes/33.md:215]: Reword here as well

File: _episodes/33.md [hunk 25]
```diff
@@ -0,0 +1,266 @@
...
+called [carbondata-presto](https://mvnrepository.com/artifact/org.apache.carbondata/carbondata-presto) 
+that works with an older version of Trino, version 333 (an io.prestosql version 
+[before the rename](https://trino.io/blog/2020/12/27/announcing-trino.html)). 
+Someone has already opened [a PR to update this connector to a current Trino version](https://github.com/apache/carbondata/pull/4198) 
+that they worked on in the middle of 2021 and hasn't made much progress 
+recently.
+
+That being said, you could build and use 
+[the Trino version of the connector](https://github.com/czy006/carbondata/tree/trino-358-alpha/integration/trino) 
+this person was working on and see if it works for you. If you are running on a
```
Comment (mosabua) [inline _episodes/33.md:229]: on, and see

**Review by mosabua — CHANGES_REQUESTED** [review]

File: _episodes/33.md (hunk 10 above)
Comment (bitsondatadev) [reply _episodes/33.md:146]: meh, they can find that out when they go to the site no?

File: _episodes/33.md [hunk 26]
```diff
@@ -0,0 +1,263 @@
+---
+layout: episode
+title:  "33: Trino becomes highly available for high demand"
+date: 2022-02-17
+tags: architecture high-availability
+youtube_id: "SEJ3UnSLE88"
+wistia_id: "w1ao9qhpi0"
+sections: 
+   - title: "Concept of the month"
+     desc: "High Availability with Trino"
```
Comment (mosabua) [inline _episodes/33.md:10]: availability

File: _episodes/33.md [hunk 27]
```diff
@@ -0,0 +1,263 @@
...
+     time: 1223
+   - title: "PR of the month"
+     desc: "PR 8956 Add support for external db for schema management in mongodb connector"
```
Comment (mosabua) [inline _episodes/33.md:13]: MongoDB

File: _episodes/33.md [hunk 28]
```diff
@@ -0,0 +1,263 @@
...
+     time: 3849
+   - title: "Bonus PR of the month"
+     desc: "PR 8202 Metadata for alias in elasticsearch connector only uses the first mapping"
+     time: 4515
+   - title: "Demo of the month"
+     desc: "Trino Fiddle: A tool for easy online testing and sharing of Trino SQL problems and their solutions"
+     time: 5528
+   - title: "Question of the month"
+     desc: "Does trino hive connector supports CarbonData?"
```
Comment (mosabua) [inline _episodes/33.md:22]: Does Trino support CarbonData?

**Review by mosabua — APPROVED** [review]
Just a few last nits.. then good to go.

**Pull request merged**